src/core/mi_memory_resource.cc

```diff
@@ -3,18 +3,25 @@
 //
 #include "core/mi_memory_resource.h"
 
+#include <sys/mman.h>
+
 #include "base/logging.h"
 
 namespace dfly {
 
-void* MiMemoryResource::do_allocate(std::size_t size, std::size_t align) {
+using namespace std;
+
+inline bool is_page_aligned(void* ptr) {
+  return (uintptr_t(ptr) & 4095) == 0;
```

Comment:
maybe add a parameter for huge pages too?

Reply:
VM pages are not part of this abstraction layer, it's something that OS manages underneath but I think you are right, if the page backing the stack is a large page, I think madvise will fail. i will check it